### ISSUE TITLE
How to model "Object maps" ?

### ISSUE THREAD

**Issue by Stwissel** [Apicurio/apicurio-studio#1064]
I can specify a property to be "Array of myType" and have something like:
`[{"name":"Peter","age":21,"Color":"blue"},{"name":"Ruth","age":19,"Color":"black"}]`

but how could I model when my JS looks like this:
```json
{
   "Peter" : {"age":21,"Color":"blue"},
   "Ruth" : {"age":19,"Color":"black"}
}
```

The [JSON spec](https://json-schema.org/understanding-json-schema/reference/object.html) suggests:

```json
{
  "type": "object",
  "properties": {  },
  "additionalProperties": { "type": "myType" }
}
```

with `myType` being:

```json
{
  "type": "object",
  "properties": {
    "age":      { "type": "number" },
    "color":    {"type": "string"}
  },
  "additionalProperties": false
}
```

How would that work (in the UI)?

**Comment by Stwissel** [comment]
`Ruth` and `Peter` are representing a flexible number of objects, not just these two. In case someone wonders why no property named after them in the schema

**Comment by EricWittmann** [comment]
Good question - this is a use case that the UI doesn't support at the moment.  You can always (of course) use the source editor as a workaround in this case.  But I'll flag this as an enhancement request and we'll need to figure out what we can do to support this in the UI.

**Comment by Stwissel** [comment]
Appreciate the swift response. Anything I can contribute here?
One way for the UI would be to add "Object Map" or "multiple named of" as label and use the same picker for the selection of the definition as used when "Array" was selected

**Comment by EricWittmann** [comment]
Yeah your suggestion is what I was thinking as well.  If you wanted to take a look at contributing that change I'd certainly be grateful for the contribution!  That said - that particular area is a little bit weird... so fair warning. :)
